mods/win7-uac-dim.wh.cpp

```diff
@@ -0,0 +1,121 @@
+// ==WindhawkMod==
+// @id              win7-uac-dim
+// @name            Windows 7 UAC Dim
+// @description     Restores the desktop screenshot on the User Account Control screen
+// @version         1.0.0
+// @author          aubymori
+// @github          https://github.com/aubymori
+// @include         consent.exe
+// @compilerOptions -lgdi32 -lgdiplus
+// ==/WindhawkMod==
+
+// ==WindhawkModReadme==
+/*
+# Windows 7 UAC Behavior
+This mod restores the full desktop screenshot on the background of the secure desktop
+for UAC. Since Windows 8, it has only displayed the wallpaper.
+
+**Before**:
+
+![Before](https://raw.githubusercontent.com/aubymori/images/main/win7-uac-dim-before.png)
+
+**After**:
+
+![After](https://raw.githubusercontent.com/aubymori/images/main/win7-uac-dim-after.png)
+*/
+// ==/WindhawkModReadme==
+
+#include <windhawk_utils.h>
+#include <gdiplus.h>
+
+using namespace Gdiplus;
+
+int x = GetSystemMetrics(SM_XVIRTUALSCREEN);
+int y = GetSystemMetrics(SM_YVIRTUALSCREEN);
+int cx = GetSystemMetrics(SM_CXVIRTUALSCREEN);
+int cy = GetSystemMetrics(SM_CYVIRTUALSCREEN);
+
+HBITMAP g_hbDesktop = NULL;
+
+void (*OnPaint_orig)(HWND) = nullptr;
+void OnPaint_hook(HWND hWnd)
+{
+    PAINTSTRUCT ps = { 0 };
+    HDC hDC = BeginPaint(hWnd, &ps);
+    HDC hMemDC = CreateCompatibleDC(hDC);
+    HBITMAP hbOld = (HBITMAP)SelectObject(hMemDC, g_hbDesktop);
+
+    BitBlt(
+        hDC,
+        0, 0,
+        cx, cy,
+        hMemDC,
+        0, 0,
+        SRCCOPY
+    );
+
+    SelectObject(hMemDC, hbOld);
+    EndPaint(hWnd, &ps);
+}
+
+const WindhawkUtils::SYMBOL_HOOK hooks[] = {
+    {
+        {
+            L"void __cdecl OnPaint(struct HWND__ *)"
+        },
+        &OnPaint_orig,
+        OnPaint_hook,
+        false
+    }
+};
+
+BOOL Wh_ModInit(void)
+{
+    /* Wait for windows and menus to go away */
+    Sleep(250);
```

Comment:
I want to preemptively say that this is probably the best you're going to get when it comes to this. Another application that captures the screen (Snipping Tool) just waits a set time before capturing the screen, and you can see the Snipping Tool window on the screenshot it takes. Likewise, you can see the window or menu UAC was launched from if it launches fast enough, even in 7. It just seems that Windows 7's UAC is slow enough that these fade away before it opens most of the time, even on real hardware. Obviously, having a half faded window or menu item on the UAC background isn't very pleasant, so I just had to do this.